client/client.go

```diff
@@ -892,9 +948,6 @@ func (c *Client) registerNode() error {
 	}
 	var resp structs.NodeUpdateResponse
 	if err := c.RPC("Node.Register", &req, &resp); err != nil {
-		if time.Since(c.start) > registerErrGrace {
```

Comment:
If you look at the comment on the `registerErrGrace`, it does have a point. It is to avoid error logs showing up in the `-dev` case. I would add it back